playbooks/ansible/roles/ctdb.setup/tasks/gpfs/main.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: Create the ctdb shared subvolume
+  include_role:
+    name: 'sit.{{ config.be.name }}'
+    tasks_from: new_volume
+  vars:
+    volume:
+      name: "ctdb"
+
+- name: Create lock directory
+  file:
+    path: "{{ config.paths.ctdb.var_lib }}/lock"
+    state: directory
+
+- name: Bind CTDB directory
+  mount:
+    path: "{{ config.paths.ctdb.var_lib }}/lock"
+    src: /gpfs/sit_fs/ctdb
+    opts: bind
+    fstype: none
+    state: mounted
```

Comment:
Yes, that seems better. However I've seen that there's no consistency on how the variables are used to indicate that the volume needs to be mounted. I'll move the mount to _new_volume.yml_ but I'll leave for a future patch normalizing how variables are used.

Reply:
On second thought, that is not so simple. _new_volume.yml_ is called once from one of the nodes because the filesystem is distributed and it only needs to be created once. However the bind mount needs to be done on all samba nodes. This is also problematic when creating a share. Ideally, the creation of a share should include the creation of the volume, but the target nodes of both operations are also different.

If you are ok, I would prefer to leave this as it is now and try to normalize it for all backends in a future change.